stock_location_tank/models/stock_picking.py

```diff
@@ -0,0 +1,165 @@
+                        % location_dest.name
+                    )
+                if len(mrp_operation_type) > 1:
+                    raise UserError(
+                        _(
+                            "More than one mrp_operation picking type for tank location %s"
+                        )
+                        % location_dest.name
+                    )
+                if not mrp_operation_type.default_location_src_id:
+                    raise UserError(
+                        _(
+                            "Not found default location source for mrp_operation picking type %s"
+                        )
+                        % mrp_operation_type.name
+                    )
+                component_quant = rec.env["stock.quant"].search(
+                    [
+                        ("product_id", "=", product.id),
+                        ("location_id", "=", location_dest.id),
+                    ]
+                )
+                quantity_to_prod = sum(component_quant.mapped("quantity"))
+                # Nosaltres agafarem la location_dest asignada al picking, no la generica asignada al picking_type
+                production = rec.env["mrp.production"].create(
+                    {
+                        "product_id": product.id,  # Required
+                        "product_qty": quantity_to_prod,  # Required
+                        "product_uom_id": product.uom_id.id,  # Required
+                        "picking_type_id": mrp_operation_type.id,  # Required
+                        "location_src_id": mrp_operation_type.default_location_src_id.id,  # Required
+                        "location_dest_id": location_dest.id,  # Required
+                        "picking_id": rec.id,
+                        "move_raw_ids": [
+                            (
+                                0,
+                                0,
+                                {
+                                    "name": product.name,  # Required
+                                    "product_id": product.id,  # Required
+                                    "location_id": mrp_operation_type.default_location_src_id.id,  # Required
+                                    "location_dest_id": location_dest.id,  # Required
+                                    "product_uom": product.uom_id.id,  # Required
+                                    "product_uom_qty": quantity_to_prod,  # Required
+                                },
+                            )
+                        ],
+                    }
+                )
+                production._onchange_move_finished_product()  # depends('product_id')
+                production._onchange_move_finished()  # depends('product_qty', 'product_uom_id')
+                production._onchange_location_dest()  # depends('location_dest_id')
+                production.action_confirm()
+                lot_lmp = rec.env["stock.production.lot"].create(
```

Comment:
Remove the `lmp`reference, it could be confusing since it's something we don't really use